roles/branch-validation/tasks/ha.yml

```diff
@@ -1,114 +1,105 @@
 ---
-## HA testing
+- name: begin ha.yml
+  debug:
+    msg: Running ha.yml playbook
+
+- name: remove previously allocated floating ip
+  os_floating_ip:
+    auth: "{{ auth }}"
+    network: external
+    server: "validate-{{ groups[instance_group][0] }}"
+    floating_ip_address: "{{ fip.floating_ip.floating_ip_address }}"
+    state: absent
+  failed_when: false
+
 - name: add floating IPs to instances on controllers
   os_floating_ip:
-    auth:
-      username: admin
-      project_name: admin
-      password: "{{ secrets.admin_password }}"
-      auth_url: "http://127.0.0.1:{{ endpoints.keystone_admin.port.backend_api }}/{{ endpoints.keystone.version }}"
+    auth: "{{ auth }}"
     network: external
     server: "validate-{{ item }}"
     reuse: true
   register: fip_controllers
   with_items: groups[ha_instance_group]
+  when: converged
+
+- name: floating ips are reachable
+  command: "ping -c 5 {{ item.floating_ip.floating_ip_address }}"
+  with_items: fip_controllers.results
+  register: pings
+  until: pings|success
+  delay: 1
+  retries: 5
+  sudo: no
+  delegate_to: localhost
+  when: converged
 
-- name: add floating ip to a non-controller instance
+- name: add floating IP to a non-controller instance
   os_floating_ip:
-    auth:
-      username: admin
-      project_name: admin
-      password: "{{ secrets.admin_password }}"
-      auth_url: "http://127.0.0.1:{{ endpoints.keystone_admin.port.backend_api }}/{{ endpoints.keystone.version }}"
+    auth: "{{ auth }}"
     network: external
     server: "validate-{{ item }}"
     reuse: true
   register: fip_compute
+  ignore_errors: yes
```

Comment:
Why would this task (which should get deleted any way given the above comment) ignore errors?